data_model/src/name.rs

```diff
@@ -18,11 +17,21 @@ use crate::{ParseError, ValidationError};
 /// [`Domain`](`crate::domain::Domain`)'s name or
 /// [`Account`](`crate::account::Account`)'s name.
 #[derive(
-    DebugCustom, Display, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Encode, Serialize, IntoSchema,
+    DebugCustom,
+    Display,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Encode,
+    Serialize,
+    IntoFfi,
```

Comment:
Also maybe I've missed that in the code, but is `FromFfi` has blanked implementation for `T: IntoFfi`?
If yes, then I think it's more convinient to derive `FromFfi` and have blanked implementation for `IntoFfi` -- just like std has `impl<T, U: From<T>> Into<U> for T`

Reply:
The situation is somewhat different. `FromFfi` does not exist, there is `TryFromReprC` which has an additional associated type. I haven't found a way to support such blanket impl at the moment. I'll refer you to my previous PR [description](https://github.com/hyperledger/iroha/pull/2314#issue-1259646200)